django/contrib/admin/views/main.py

```diff
@@ -274,31 +273,15 @@ def get_filters(self, request):
         except FieldDoesNotExist as e:
             raise IncorrectLookupParameters(e) from e
 
-    def get_query_string(self, new_params=None, remove=None):
-        if new_params is None:
-            new_params = {}
-        if remove is None:
-            remove = []
-        p = self.filter_params.copy()
-        for r in remove:
-            for k in list(p):
-                if k.startswith(r):
-                    del p[k]
-        for k, v in new_params.items():
-            if v is None:
-                if k in p:
-                    del p[k]
-            else:
-                p[k] = v
-        return "?%s" % urlencode(sorted(p.items()), doseq=True)
-
     def get_results(self, request):
-        paginator = self.model_admin.get_paginator(
-            request, self.queryset, self.list_per_page
+        pagination = Pagination(
```

Comment:
Calling `Pagination` directly seems less flexible than having a separate `get_paginator` method.

If feels like the recommended approach from above  (`get_pagination_class` + `get_pagination` would be more flexible and extendable)